internal/bminventory/inventory.go

```diff
@@ -1847,20 +1847,19 @@ func (b *bareMetalInventory) updateNonDhcpNetworkParams(updates map[string]inter
 	var err error
 	*machineCidr, err = network.CalculateMachineNetworkCIDR(apiVip, ingressVip, cluster.Hosts)
 	if err != nil {
-		log.WithError(err).Errorf("failed to calculate machine network cidr for cluster: %s", params.ClusterID)
-		return common.NewApiError(http.StatusBadRequest, err)
+		return nil
```

Comment:
Why do you return nil when you have an error?
You need to skip this validation, not ignore the error.

Reply:
this is not only a validation, but it's also finding a CIDR in none-DHCP mode, I thought we just don't want to validate it if provided

Reply:
If it's needed then split the logic, don't hide errors.